### PR TITLE
Support schemaless channels

### PR DESCRIPTION
**Pull request by jtbandes**
**User-Facing Changes**
Added support for loading data from MCAP channels with no schema.

**Description**
Allow schema/schemaName to be undefined in several places. Remove some error messages around missing schemas.

Resolves FG-1441

### CODE REVIEW

File: packages/mcap-support/src/parseChannel.ts [hunk 1]
```diff
@@ -49,33 +49,31 @@ function parsedDefinitionsToDatatypes(
  */
 export function parseChannel(channel: Channel): ParsedChannel {
   if (channel.messageEncoding === "json") {
-    if (channel.schema?.encoding !== "jsonschema") {
+    if (channel.schema != undefined && channel.schema.encoding !== "jsonschema") {
```
Comment (jhurliman) [inline packages/mcap-support/src/parseChannel.ts:52]: I thought our eslint rules would enforce the `?.encoding` variant here. Was this change done by linting or manually?

File: packages/studio-base/src/players/IterablePlayer/Mcap/McapIndexedIterableSource.ts [hunk 2]
```diff
@@ -191,7 +184,7 @@ export class McapIndexedIterableSource implements IIterableSource {
             publishTime: fromNanoSec(message.publishTime),
             message: channelInfo.parsedChannel.deserializer(message.data),
             sizeInBytes: message.data.byteLength,
-            schemaName: channelInfo.schemaName,
+            schemaName: channelInfo.schemaName ?? "",
```
Comment (jhurliman) [inline packages/studio-base/src/players/IterablePlayer/Mcap/McapIndexedIterableSource.ts:187]: :( maybe it should be `"-"` to match the UI 🤣

**Review by jhurliman — APPROVED** [review]

File: packages/mcap-support/src/parseChannel.ts (hunk 1 above)
Comment (jtbandes) [reply packages/mcap-support/src/parseChannel.ts:52]: Manually but the linter didn't complain. I'm always a bit surprised about when that rule fires and when it doesn't 🤷

File: packages/studio-base/src/components/PanelExtensionAdapter/renderState.ts [hunk 3]
```diff
@@ -171,8 +171,8 @@ function initRenderStateBuilder(): BuildRenderStateFn {
         const topics = sortedTopics.map<Topic>((topic) => {
           const newTopic: Topic = {
             name: topic.name,
-            datatype: topic.schemaName,
-            schemaName: topic.schemaName,
+            datatype: topic.schemaName ?? "",
+            schemaName: topic.schemaName ?? "",
```
Comment (snosenzo) [inline packages/studio-base/src/components/PanelExtensionAdapter/renderState.ts:175]: what's the general rule for where this needs to be a string vs string | undefined?
Should they still not show externally as `string | undefined`?

**Review by snosenzo — APPROVED** [review]
just a question about when they shouldn't be able to be undefined

File: packages/studio-base/src/components/PanelExtensionAdapter/renderState.ts (hunk 3 above)
Comment (jtbandes) [reply packages/studio-base/src/components/PanelExtensionAdapter/renderState.ts:175]: @defunctzombie suggested that we should not change the public APIs in @foxglove/studio which means we can't change the type of MessageEvent.

File: packages/mcap-support/src/parseChannel.ts [hunk 4]
```diff
@@ -49,33 +49,31 @@ function parsedDefinitionsToDatatypes(
...
       throw new Error(
-        `Message encoding ${channel.messageEncoding} with ${
-          channel.schema == undefined
-            ? "no encoding"
-            : `schema encoding '${channel.schema.encoding}'`
-        } is not supported (expected jsonschema)`,
+        `Message encoding ${channel.messageEncoding} with schema encoding '${channel.schema.encoding}' is not supported (expected jsonschema or no schema)`,
```
Comment (defunctzombie) [inline packages/mcap-support/src/parseChannel.ts:54]: Where does the check happen that schemaless uses a schemId of `0`? Or are we saying that users can still provide a schema record with no schema for json encoded messages? Technically not a valid mcap file (or maybe it is?) but we want to be flexible about it in studio?

**Review by defunctzombie — APPROVED** [review]

File: packages/mcap-support/src/parseChannel.ts (hunk 4 above)
Comment (jtbandes) [reply packages/mcap-support/src/parseChannel.ts:54]: Does this answer your question? https://github.com/foxglove/studio/pull/5351/files/d42e95b3e69bcfa577ffd96b589c4e2b968bc6e0#diff-2a8df5991c9755a3c42c61399d45b12e2e04c804b97b780053696725b984f7d8R55

Comment (defunctzombie) [reply packages/mcap-support/src/parseChannel.ts:54]: Yep 👍

**Pull request merged**